src/main/java/it/cosenonjaviste/security/jwt/utils/JwtTokenBuilder.java

```diff
@@ -101,8 +101,8 @@ public static JwtTokenBuilder from(String token, String secret) {
 	 */
 	public static JwtTokenBuilder from(JwtTokenVerifier verifier, String secret) {
 		JwtTokenBuilder builder = create(secret);
-		Map<String, Object> verifiedClaims = verifier.getClaims();
-		restoreInternalStatus(builder, verifiedClaims);
+		DecodedJWT decodedJWT = verifier.getDecodedJWT();
```

Comment:
Good point! I made a test to verify and you're right, algorithm from verifier is lost!
I deprecated the method, fixed its behavior and created a new overload without secret parameter because is not needed anymore.

Reply:
I deprecated almost all overloads of `from`, never used internally and should not be used. Please check my last commit